org.eclipse.jdt.core.tests.builder/src/org/eclipse/jdt/core/tests/builder/ParticipantBuildTests.java

```diff
@@ -405,6 +405,110 @@ public void processAnnotations(BuildContext[] files) {
 		expectingNoProblems();
 	}
 
+	public void testPostProcessing() throws JavaModelException {
```

Comment:
We need **another** test that tests that three participants configured with "requires" are:
1) Executed in expected order
2) See the output of each other
3) the final class bytes contain results of all three participants
4) Participants can add one more class to be recompiled via "dependencies" set on result
5) Participants can create extra compilation problems.

Reply:
> We need **another** test that tests that three participants configured with "requires" are:
> 
>     1. Executed in expected order
> 
>     2. See the output of each other
> 
>     3. the final class bytes contain results of all three participants
> 
>     4. Participants can add one more class to be recompiled via "dependencies" set on result
> 
>     5. Participants can create extra compilation problems.

@iloveeclipse 
Is there some way to register compilation participants on the fly? I would like to prevent to register them via plugin.xml since they would be visible for all tests if we need it just for one, but not sure if this is possible.

Reply:
> Is there some way to register compilation participants on the fly

No, I don't see a way right now.
See 
- final org.eclipse.jdt.internal.core.JavaModelManager.compilationParticipants
- org.eclipse.jdt.internal.core.JavaModelManager.CompilationParticipants.getRegisteredParticipants()

But in theory the "dummy" test participants shouldn't harm if they are added in a way where they only work during test.

Reply:
`CompilationParticipant.isActive(IJavaProject)` should just return false for any project except the one used in the test.